### PR TITLE
Sprint: Anderson

### PR DESCRIPTION
**Pull request by brizandrew**
DO NOT MERGE YET

The start of an atoms rework. We should wait until we have everything before merging, but I think it's at a good enough to start a review. Don't worry about making sure everything works or running it, we'll worry about that when we actually start doing some integration. Know that's its been tested _enough_ to prove itself. If there's edge cases you can think of that aren't being handled feel free to note.

Also note that `atoms-legacy` and `actions-legacy` are two directories that don't need to be reviewed. They're there as an easy reference but will be deleted before a final merge.

### CODE REVIEW

File: src/store/constants.js
Comment (aaronalbright) [inline src/store/constants.js:1]: What's the thinking behind naming these `-new` ? To avoid conflicts with our current stores?

Comment (brizandrew) [reply src/store/constants.js:1]: Yup, that's temporary, I should add some TODO flags. Basically just so I can go between branches and not have to worry about my story being messed up.

File: src/utils/entriesToObject.js
Comment (aaronalbright) [inline src/utils/entriesToObject.js:1]: Mentioned earlier but I'm almost certain this does the same thing that `Object.fromEntries()` does? With a small performance boost as well.

File: src/atoms/illustrations/index.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import atom from './atom';
+
+import onReadEntities from './sync/read';
+import onWriteEntities from './sync/write';
```
[the remark this answers is not in the document]
Comment (brizandrew) [reply src/atoms/illustrations/index.js:4]: Good catch, this whole file should be deleted

File: src/atoms/projects/write.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import store from '../../store';
+
+/**
+ * Writes new data to a project entity
+ * @function
+ * @param {Object} value - The data to update with
+ * @param {String=} value.name - A new name
+ * @param {String=} value.slug - A new slug
+ * @param {String=} value.lastUpdated - A new last updated time
```
[the remark this answers is not in the document]
Comment (brizandrew) [reply src/atoms/projects/write.js:9]: I thought that's how you signal that it's optional? That's what I meant to do here.

File: src/atoms/settings/read.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+import entriesToObject from '../../utils/entriesToObject';
+
+/**
+ * Gets the user's active settings
+ * @function
+ * @returns {Object}
+ */
+export default async function onReadSettings() {
+  const settingsEntries = await store.settings.get();
+  return entriesToObject(settingsEntries);
```
[the remark this answers is not in the document]
Comment (brizandrew) [reply src/atoms/settings/read.js:11]: Lol nope you right. I just didn't know that existed. Will change.

**Pull request closed without merging**